lib/Manager.php

```diff
@@ -341,6 +341,39 @@ public function getRoomsForUser(string $userId, bool $includeLastMessage = false
 		return $rooms;
 	}
 
+	/**
+	 * @param string $userId
+	 * @return string[]
+	 */
+	public function getRoomTokensForUser(string $userId): array {
+		$query = $this->db->getQueryBuilder();
+		$query->select('token')
+			->from('talk_rooms', 'r')
+			->leftJoin('r', 'talk_attendees', 'a', $query->expr()->andX(
+				$query->expr()->eq('a.actor_id', $query->createNamedParameter($userId)),
+				$query->expr()->eq('a.actor_type', $query->createNamedParameter(Attendee::ACTOR_USERS)),
+				$query->expr()->eq('a.room_id', 'r.id')
+			))
+			->leftJoin('a', 'talk_sessions', 's', $query->expr()->andX(
+				$query->expr()->eq('a.id', 's.attendee_id')
+			))
+			->where($query->expr()->isNotNull('a.id'));
```

Comment:
In this case it might even make more sense to select from this table and left join the rooms instead